banking/src/main/java/rocks/kata/Account.java

```diff
@@ -0,0 +1,61 @@
+package rocks.kata;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.LongAdder;
+
+import rocks.kata.AccountOperation.OperationType;
+import rocks.kata.DatabaseClasses.AccountOperationsDatabase;
+
+public class Account {
+
+    private static final LongAdder numberOfAccounts = new LongAdder();
+
+    private AtomicInteger balance;
+    private long uniqueAccountID;
+    private AccountOperationsDatabase operationsDatabase;
+
+    public Account(AccountOperationsDatabase operationsDatabase) {
+        numberOfAccounts.increment();
+        this.uniqueAccountID = numberOfAccounts.longValue();
```

Comment:
Rozbicie na 2 operacje sprawiają, ze ta operacja nie jest już atomowa. Dwa różne wątki mogą wykonać linijkę 18. w tym samym czasie, a następnie oba przypisać sobie to samo ID. `AtomicLong` i inne atomiki mają na takie okazje `incrementAndGet()`, który pozwala zamienić te 2 opracje na jedną, atomową.

Reply:
Indeed, zjebałam aż mi wstyd